tests/unit/artifact/test_koji.py

```diff
@@ -30,16 +29,26 @@ def test_call_api_success(koji_provider):
     assert result == "ok"
 
 
-def test_fetch_rpms_populates_list(root_logger):
-    rpm_data = [{"name": "foo", "version": "1.x", "release": "1", "nvr": "foo-1.x-1"}]
-
-    with patch.object(KojiArtifactProvider, "_call_api", return_value=rpm_data) as mock_call:
-        provider = KojiArtifactProvider(root_logger, "koji.build:123")
-        assert provider._rpm_list == rpm_data
-        mock_call.assert_called_once_with("listBuildRPMs", 123)
+def test_list_artifacts_returns_rpm_artifacts(koji_provider):
+    koji_provider._rpm_list = [
+        {"nvr": "foo-1.0-1", "arch": "x86_64", "name": "foo", "version": "1.0", "release": "1"}
+    ]
+    artifacts = list(koji_provider.list_artifacts())
+    assert len(artifacts) == 1
+    artifact = artifacts[0]
+    assert artifact.id.startswith("foo-1.0-1")
+    assert "kojipkgs.fedoraproject.org" in artifact.location
```

Comment:
These checks should be more narrow so that we are testing for any regressions in the formats